### ISSUE TITLE
Memory is not released after running the melt function

### ISSUE THREAD

**Issue by Chuck321123** [pola-rs/polars#12629]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
from memory_profiler import profile
print("After importing memory profiler")

@profile
def my_function():
    
    import polars as pl
    import numpy as np
    from datetime import datetime, timedelta
    import time
    
    # Set seed for reproducibility
    np.random.seed(42)
    
    # Number of rows
    num_rows = 10000
    
    # Generate datetime values within a date range
    start_date = datetime(2023, 1, 1)
    end_date = datetime(2023, 1, 5)
    date_values = [start_date + timedelta(days=np.random.uniform(0, 5)) for _ in range(num_rows)]
    
    # Create a polars DataFrame
    df = pl.DataFrame({
        'Datetime': date_values,
        **{f'Col_{i}': np.random.rand(num_rows) for i in range(1, 101)}
    })
    
    df = (df.melt(id_vars=["Datetime"], value_vars=df.columns[1:], variable_name="Name", value_name="Values").filter(pl.col("Values").is_not_null()))
    
    del(df)
    
    # Some random operations
    e=1
    time.sleep(4)
    f=2
    for i in range(1000):
        g=i
        
    lst = [None] * 10000

my_function()
```

### Issue description

So before using polars, my code would stabilize around 190-200Mb of ram, after switching to polars, my code uses about 300Mb of RAM, which is far more than what i use. The true answer after running sum(sys.getsizeof(obj) for obj in gc.get_objects()) / (1024 * 1024) is that i only use about 30mb of objects + 150mb of imported packages, which there is obviously memory not being released to the system. After running memory-profiler several times in different occasions, it constantly happens after i run a melt function and maybe after converting to pandas, which i kind of expect, but not from melting. Even after deleting the melted objects (and every other big object), the ram is still not released. Please see attached screenshot and code. Hoping for a reply
![Memory not being released](https://github.com/pola-rs/polars/assets/123987501/10c941a4-0bb3-4461-9442-ba65c24e3c59)

### Expected behavior

Memory is being released back to the system

### Installed versions

<details>

```
cloudpickle = 3.0.0
matplotlib=3.8.1
numpy=1.26.2
openpyxl=3.1.2
pandas=2.1.3
pyarrow=14.0.1
polars=0.19.15

```

</details>

**Comment by ritchie46** [comment]
Please fill in the installed versions properly.

**Comment by Chuck321123** [comment]
@ritchie46 Done.

**Comment by ritchie46** [comment]
Yes, the memory allocator keeps your memory around for later use cases. This is not a bug, but how memory allocators work.

Though, I am considering to drop mimalloc for windows and use the default allocator.

Could you try if these help? https://microsoft.github.io/mimalloc/environment.html

For instance `MIMALLOC_RESET_DELAY=25` instead of `100`.

**Comment by Chuck321123** [comment]
@ritchie46 Didn't work. Problem is also that several minutes later (in a while True loop), the ram is still not released.

**Comment by ritchie46** [comment]
> several minutes later (in a while True loop), the ram is still not released.

That's expected. The memory allicator keeps it around for later usage.

**Comment by Chuck321123** [comment]
@ritchie46 Well, is there no settings that can be implemented to release this ram? Quite unfortunate when you have several scripts running at the same time on a Raspberry Pi and you run out of memory

**Comment by ritchie46** [comment]
Please read this answer on SO. If you do a lot of heap allocations, you must accept memory being kept by a process: https://stackoverflow.com/a/45539066/6717054

**Comment by Chuck321123** [comment]
@ritchie46 I see. Unfortunately, the "problem" also exists on Raspberry Pi on Ubuntu.

**Comment by ritchie46** [comment]
#27797
